### PR TITLE
[ADAM-1694] Add short readable descriptions for toString in subclasses of GenomicRDD.

### LINKED ISSUE #1694: Make GenomicRDD.toString more adam-shell friendly

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1694

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala [hunk 1]
```diff
@@ -490,4 +490,7 @@ sealed abstract class NucleotideContigFragmentRDD extends AvroGenomicRDD[Nucleot
       (p._1, p._2: java.lang.Long)
     }).toJavaRDD()
   }
+
+  override def toString = "%s with %d reference sequences"
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala]: You should be able to factor this into the `GenomicRDD` trait, no?

Comment (heuermh) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala]: I think it is possible ... but I read some odd things about toString on traits and figured the sealed abstract classes would be the best place.  Then the messages are slightly different so it isn't exactly breaking DRY.

Comment (fnothaft) [reply adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala]: What odd things did you see?

**Comment by fnothaft** [comment]
Merged! Thanks @heuermh!

**Pull request merged**